bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/PageImpl.java

```diff
@@ -146,4 +151,12 @@ public String getCssClassNames() {
     public NavigationItem getRedirectTarget() {
         return redirectTarget;
     }
+
+    @Override
+    public boolean hasCloudconfigSupport() {
+        if (productInfoProvider == null) {
+            return false;
+        }
+        return productInfoProvider.getProductInfo().getVersion().compareTo(new Version("6.4.0")) >= 0;
```

Comment:
I would also check for non `null` for `productInfoProvider.getProductInfo()` and `productInfoProvider.getProductInfo().getVersion()`.

Reply:
Not sure if these will ever return null; at least in the API docs it's not mentioned.

Reply:
Exactly, better safe than sorry :)